### PR TITLE
Kazantsev Evgeny Lab 4 Var 3 

### PR DESCRIPTION
**Pull request by neotouch1**

### CODE REVIEW

File: mlir/test/Transforms/lab4/kazantsev_evgeny/test.mlir [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// RUN: split-file %s %t
```
Comment (aobolensk) [inline mlir/test/Transforms/lab4/kazantsev_evgeny/test.mlir:1]: Come up with your own tests

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: mlir/test/Transforms/lab4/kazantsev_evgeny/test.mlir (hunk 1 above)
Comment (neotouch1) [reply mlir/test/Transforms/lab4/kazantsev_evgeny/test.mlir:1]: Added my own tests

**Review by m-ly4 — APPROVED** [review]

File: mlir/lib/Transforms/lab4/kazantsev_evgeny/KazantsevPassDepthMax.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+namespace {
+class KazantsevPassDepthMax
+    : public PassWrapper<KazantsevPassDepthMax, OperationPass<func::FuncOp>> {
+public:
+  StringRef getArgument() const final { return "max-depth-pass"; }
+  StringRef getDescription() const final {
+    return "Counts the max depth of region nests in a function.";
+  }
+  std::map<Operation *, int> Depth;
```
Comment (m-ly4) [inline mlir/lib/Transforms/lab4/kazantsev_evgeny/KazantsevPassDepthMax.cpp:21]:
```suggestion
  llvm::DenseMap<Operation *, int> Depth;
```

**Review by m-ly4 — CHANGES_REQUESTED** [review]

**Review by aobolensk — APPROVED** [review]

**Comment by m-ly4** [comment]
@neotouch1 Could you please apply my remark and after that I'll merge your patch into master.

**Review by m-ly4 — APPROVED** [review]

**Pull request merged**